server/game/GameActions/LeavePlay.js

```diff
@@ -0,0 +1,50 @@
+const GameAction = require('./GameAction');
+const ReturnCardToHand = require('./ReturnCardToHand');
+const DiscardCard = require('./DiscardCard');
+
+class LeavePlay extends GameAction {
+    constructor() {
+        super('leavePlay');
+    }
+
+    canChangeGameState({ card }) {
+        return ['active plot', 'faction', 'play area', 'title'].includes(card.location);
+    }
+
+    createEvent({ card, allowSave = false }) {
+        const params = {
+            card,
+            allowSave,
+            automaticSaveWithDupe: true
+        };
+
+        return this.event('onCardLeftPlay', params, event => {
+            event.cardStateWhenLeftPlay = event.card.createSnapshot();
```

Comment:
IIRC we wanted to seperate the snapshot creation from the event handler like we did for `DiscardCard.js` so interrupts that overwrite the handler don´t overwrite the snapshot creation. I suggest adding `snapshotName: 'cardStateWhenLeftPlay'` to the params and then the snapshot will be created automatically inside `event.js`

Reply:
Good catch, this class was originally put together before you had added that mechanism.